### PR TITLE
Prometheus: Update onblur ref with value of last changed monaco editor

### LINKED ISSUE #55387: Query does not run with Shift+Enter, but is run by clicking "Run query"
**What happened**:
When I have a PromQL or LogQL query and I hit `<Shift>+<Enter>` to execute the query, occasionally the query in the editor is not sent to the server. When I click the `Run query` button, the query in the editor is run as expected.

**What you expected to happen**:
I expect the query to be sent as written from the editor to the server when hitting `<Shift>+<Enter>` every time.

**How to reproduce it (as minimally and precisely as possible)**:
Not exactly sure how to reproduce precisely. But it occurs nearly every time I use Grafana.

**Anything else we need to know?**:

**Environment**:
- Grafana version: v9.2.0-33764pre (18f33871d1)
- Data source type & version: PromQL
- OS Grafana is installed on: Linux
- User OS & Browser: Linux, Chrome
- Grafana plugins:
- Others:

### PR DESCRIPTION
**Pull request by gtk-grafana**
Add new monaco callback to update onBlurRef with current value on monaco editor change.

**What this PR does / why we need it**:
Fixes bug when submitting queries via the keyboard when using multiple queries in prometheus. Currently only the last editor input that is un-blurred will get run when the user presses shift+enter, so if you make changes in a query editor, and press shift+enter without clicking on something else first, the updates to your query will not be sent to the datasource API, and you will instead see the results of the old (unchanged) query populating the UI.

**Which issue(s) this PR fixes**:
https://github.com/grafana/grafana/issues/55387

Fixes https://github.com/grafana/grafana/issues/55387 

**Special notes for your reviewer**:
What am I breaking this time?

### CODE REVIEW

**Review by bohandley — COMMENTED** [review]
*edit - this is from panels, not explore 

I think I don't quite understand how the onBlurRef works in the Monaco editor of Prometheus. With this implementation it looks like the query is being run on every key stroke now, which is not optimal. 
Screen Shot 2022-09-20 at 3 06 45 PM

**Comment by gtk-grafana** [comment]
Ok, so this bug only appears to be active on the explore view, and while this fix does fix the explore view, it causes the panel view to query on every keystroke. Thanks @bohandley I'm pulling this back into draft and gonna play around a bit longer

**Comment by bohandley** [comment]
Thanks @gtk-grafana for looking in to this!

**Comment by gtk-grafana** [comment]
Ok, so this is now working for me in explore and dashboard views, the trouble with my initial implementation is I didn't handle the case where we run a query for every onblur event (dashboard context), since we were overloading the onBlur with onChange, each onChange triggered another API query. Fixing this by just updating the query without updating the onblur by introducing new `onChange` handler to MonacoQueryFieldWrapper

**Review by bohandley — APPROVED** [review]
Sweet! I tested with panels and explore and I can find no bugs. LGTM!

File: public/app/plugins/datasource/prometheus/components/monaco-query-field/MonacoQueryField.tsx [hunk 1]
```diff
@@ -201,6 +202,12 @@ const MonacoQueryField = (props: Props) => {
           editor.onDidContentSizeChange(updateElementHeight);
           updateElementHeight();
 
+          // Whenever the editor changes, lets save the last value so the next query for this editor will be up-to-date.
+          // This change is being introduced to fix a bug where you can submit a query via shift+enter, but if you just clicked into another field and haven't un-blurred the active field, then the query that is run will be stale, as the reference is only updated with the value of the last blurred input.
```
Comment (itsmylife) [inline public/app/plugins/datasource/prometheus/components/monaco-query-field/MonacoQueryField.tsx:206]:
```suggestion
          // Whenever the editor changes, lets save the last value so the next query for this editor will be up-to-date.
          // This change is being introduced to fix a bug where you can submit a query via shift+enter, 
          // but if you just clicked into another field and haven't un-blurred the active field, 
          // then the query that is run will be stale, as the reference is only updated 
          // with the value of the last blurred input.
```

Comment (itsmylife) [reply public/app/plugins/datasource/prometheus/components/monaco-query-field/MonacoQueryField.tsx:206]: It was just breaking down the long comment line :)

**Review by itsmylife — APPROVED** [review]

**Comment by gtk-grafana** [comment]
Closing/Reopening to trigger drone build

**Pull request merged**